client/driver/docker.go

```diff
@@ -1139,8 +1139,9 @@ func (d *DockerDriver) createContainerConfig(ctx *ExecContext, task *structs.Tas
 
 	// Calculate CPU Quota
 	if driverConfig.CPUHardLimit {
+		numCores := runtime.NumCPU()
```

Comment:
Good catch! It definitely seems like the period needs to be multiplied per core:

> Note that the quota and period parameters operate on a CPU basis. To allow a process to fully utilize two CPUs, for example, set cpu.cfs_quota_us to 200000 and cpu.cfs_period_us to 100000.

via https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/resource_management_guide/sec-cpu

Mind adding a comment like:

```go
// cfs_quota_us is the time per core, so we must 
// multiply the time by the number of cores available
// See https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/resource_management_guide/sec-cpu
```